src/ext-test/java/org/opentripplanner/ext/emissions/EmissionsTest.java

```diff
@@ -0,0 +1,188 @@
+import org.opentripplanner.transit.model._data.TransitModelForTest;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.Deduplicator;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.organization.Agency;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.model.timetable.TripTimes;
+
+class EmissionsTest {
+
+  private EmissionsDataService eService;
+  private EmissionsFilter emissionsFilter;
+
+  static final ZonedDateTime TIME = OffsetDateTime
+    .parse("2023-07-20T17:49:06+03:00")
+    .toZonedDateTime();
+
+  private static final Agency subject = Agency
+    .of(TransitModelForTest.id("F:1"))
+    .withName("Foo_CO")
+    .withTimezone("Europe/Helsinki")
+    .build();
+
+  @BeforeEach
+  void SetUp() {
+    Map<FeedScopedId, Double> emissions = new HashMap<>();
+    emissions.put(new FeedScopedId("F", "1"), (0.12 / 12));
+    emissions.put(new FeedScopedId("F", "2"), 0.0);
+    EmissionsDataModel emissionsDataModel = new EmissionsDataModel();
+    emissionsDataModel.setCo2Emissions(emissions);
+    emissionsDataModel.setCarAvgCo2PerMeter(0.131);
+    this.eService = new EmissionsDataService(emissionsDataModel);
+    this.emissionsFilter = new EmissionsFilter(eService);
+  }
+
+  @Test
+  void testGetEmissionsForItinerary() {
+    var stopOne = TransitModelForTest.stopForTest("1:stop1", 60, 25);
+    var stopTwo = TransitModelForTest.stopForTest("1:stop1", 61, 25);
+    var stopThree = TransitModelForTest.stopForTest("1:stop1", 62, 25);
+    var stopPattern = TransitModelForTest.stopPattern(stopOne, stopTwo, stopThree);
+    var route = TransitModelForTest.route(id("1")).build();
+    List<Leg> legs = new ArrayList<>();
+    var pattern = TransitModelForTest.tripPattern("1", route).withStopPattern(stopPattern).build();
+    var stoptime = new StopTime();
+    var stoptimes = new ArrayList<StopTime>();
+    stoptimes.add(stoptime);
+    var trip = Trip
+      .of(FeedScopedId.parse("FOO:BAR"))
+      .withMode(TransitMode.BUS)
+      .withRoute(route)
+      .build();
+    var leg = new ScheduledTransitLeg(
+      new TripTimes(trip, stoptimes, new Deduplicator()),
+      pattern,
+      0,
+      2,
+      TIME,
+      TIME.plusMinutes(10),
+      TIME.toLocalDate(),
+      ZoneIds.BERLIN,
+      null,
+      null,
+      100,
+      null
+    );
+    legs.add(leg);
+    Itinerary i = new Itinerary(legs);
+    assertEquals(2223.902, emissionsFilter.getEmissionsForItinerary(i, EmissionType.CO2).get());
+  }
+
+  @Test
+  void testGetEmissionsForCarRoute() {
+    List<Leg> legs = new ArrayList<>();
+    var leg = StreetLeg
+      .create()
+      .withMode(TraverseMode.CAR)
+      .withDistanceMeters(214.4)
+      .withStartTime(TIME)
+      .withEndTime(TIME.plus(1, ChronoUnit.HOURS))
+      .build();
+    legs.add(leg);
+    Itinerary i = new Itinerary(legs);
```

Comment:
```suggestion
    var leg = StreetLeg
      .create()
      .withMode(TraverseMode.CAR)
      .withDistanceMeters(214.4)
      .withStartTime(TIME)
      .withEndTime(TIME.plus(1, ChronoUnit.HOURS))
      .build();
    Itinerary i = new Itinerary(List.of(legs));
```

Reply:
No need to create mutable lists, the `List.of()` constructor is super-convenient.